src/ducks/auth.ts

```diff
@@ -1701,5 +1727,58 @@ export const useAuthenticationStore = create<AuthStore>()((set, get) => {
       password: string,
       activeAccountId?: string | null,
     ) => getKeyFromKeyManager(password, activeAccountId),
+
+    importSecretKey: async (secretKey: string, password: string) => {
+      set({ isLoading: true, error: null });
+
+      try {
+        const keypair = Keypair.fromSecret(secretKey);
+        const publicKey = keypair.publicKey();
+        const privateKey = keypair.secret();
+
+        const existingAccounts = await getAllAccounts();
+        const accountExists = existingAccounts.some(
+          (account) => account.publicKey === publicKey,
+        );
+
+        if (accountExists) {
+          throw new Error(t("authStore.error.accountAlreadyExists"));
+        }
+
+        const temporaryStore = await getTemporaryStore();
+        if (!temporaryStore) {
+          throw new Error(t("authStore.error.temporaryStoreNotFound"));
+        }
+
+        const keyPair = {
+          publicKey,
+          privateKey,
+        };
+
```

Comment:
```suggestion
```

Reply:
I'm suggesting removing it from here and making it inline since we already have a `keypair` const on line 1735 so it looks a bit less confusing